### PR TITLE
CMOR fixes for HadREM3-GA7-05

### LINKED ISSUE #1044: Dataset problem: CMORCheckError for MOHC-HadREM3-GA7-05 CORDEX data. (Wrong latitude and longitude names)
Data from the Met Office run CORDEX EUR-11 domain experiments fail CMOR checking:

> ERROR   Failed to run fix_metadata([<iris 'Cube' of air_temperature / (K) (time: 120; grid_latitude: 412; grid_longitude: 424)>, <iris 'Cube' of air_temperature / (K) (time: 120; grid_latitude: 412; grid_longitude: 424)>, <iris 'Cube' of air_temperature / (K) (time: 120; grid_latitude: 412; grid_longitude: 424)>], {'project': 'CORDEX', 'dataset': 'MOHC-HadREM3-GA7-05', 'short_name': 'tas', 'mip': 'mon', 'frequency': 'mon', 'check_level': <CheckLevels.DEFAULT: 3>})

> esmvalcore.cmor.check.CMORCheckError: There were errors in variable tas:
Coordinate longitude has var name longitude instead of lon
 Coordinate latitude has var name latitude instead of lat

- Data file has been changed by you in any way (if you answer yes the issue will be void and closed, we are not
supporting custom-made datasets that present problems, it is your resposability to fix those problems): No
- Project (CMIP5/CMIP6/OBS/obs4mips/etc): CORDEX
- Full dataset description (fill out as many as you know, please):
  - dataset: MOHC-HadREM3-GA7-05
  - experiment: historical and rcp85
  - mip: mon
  - grid: N/A
  - type:
  - version: v2
  - tier: N/A
  - variable used: tas
  - data version:
- Problems encountered (please describe what the actual problems are: e.g. wrong standard name, issue with dimensions): Wrong dimension names.
- Pointer to existing copy of data on ESGF node (it would be very useful if you could provide a physical
fullpath to the file(s) that are causing the problem, e.g. on CEDA Jasmin or DKRZ): /badc/cordex/data/CORDEX/output/EUR-11/MOHC/CNRM-CERFACS-CNRM-CM5/historical/r1i1p1/MOHC-HadREM3-GA7-05/v2/mon/tas/latest/tas_EUR-11_CNRM-CERFACS-CNRM-CM5_historical_r1i1p1_MOHC-HadREM3-GA7-05_v2_mon_199101-200012.nc
[main_log_debug.txt](https://github.com/ESMValGroup/ESMValCore/files/6131139/main_log_debug.txt)
[ncdump.txt](https://github.com/ESMValGroup/ESMValCore/files/6131140/ncdump.txt)
[recipe_cordex_test.yml.txt](https://github.com/ESMValGroup/ESMValCore/files/6131152/recipe_cordex_test.yml.txt)
(txt extension added since github wouldn't let me upload a .yml file ¯\_(ツ)_/¯

- Other notes and mentions: To my knowledge this appears to affect all data under MOHC in the EUR-11 domain. I suspect this may affect all MO RCM data and is related to the CMORization settings that have been used in internal CMORization suites, since I see the same errors working internally with data from the EUCP CPM experiments (they have not been uploaded to ESGF yet, but are planned to be), and also saw the same behaviour with data from own Caribbean RCM model experiments that were CMORized at the Met Office. Recipes using these datasets work without issue by setting `check_level=relaxed` when running

**Assign the right people**
If you are already a member of the ESMValTool GitHub project, please assign Valeriu Predoi (valeriupredoi) and
Klaus Zimmermann (zklaus) to the issue. They will then check the issue raised and propagate it further to the
data model developers.
@valeriupredoi @zklaus 

**Please attach**
  - The recipe that you are trying to run, you can find a copy in the `run` directory in the output directory
  - The `main_log_debug.txt` file, this can also be found in the `run` directory in the output directory
  - Attach a text file with the output of `ncdump -h /path/to/file.nc`

Cheers :beer:

### PR DESCRIPTION
**Pull request by thomascrocker**
CMOR fixes for HadREM3-GA7-05 CORDEX data. Follows similar approach to fixes for HadGEM2-ES etc.

-   Closes #1044

***

## Before you get started

-   [X] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## Checklist

-   [X] PR has a descriptive title for the [changelog](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [X] Labels are assigned so they can be used in the [changelog](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [X] Code follows the [style guide](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#code-style)
-   [ ] [Documentation](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#documentation) is available for new functionality
-   [ ] YAML files pass [`pre-commit`](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#pre-commit) or [`yamllint`](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/community/introduction.html#yaml) checks
-   [ ] [Circle/CI tests pass](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [ ] [Codacy code quality checks pass](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review)
-   [ ] [Documentation builds successfully](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#branches-pull-requests-and-code-review) on readthedocs
-   [ ] [Unit tests](https://docs.esmvaltool.org/projects/esmvalcore/projects/esmvalcore/en/latest/contributing.html#contributing-to-the-esmvalcore-package) are available

If you make backwards incompatible changes to the recipe format:

-   [ ] Update [ESMValTool](https://github.com/esmvalgroup/esmvaltool) and link the pull request(s) in the description

***

To help with the number pull requests:

-   🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValTool/pulls) in this repository

### CODE REVIEW

**Comment by thomascrocker** [comment]
My first pull request. Feel free to assign me to review any other PRs if it would be useful.

**Comment by valeriupredoi** [comment]
cheers @thomascrocker I'll have a looksee later today :+1:

File: esmvalcore/cmor/_fixes/cordex/hadrem3_ga7_05.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+"""Fix HadREM3-GA7.05"""
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cordex/hadrem3_ga7_05.py:1]:
```suggestion
"""Fix HadREM3-GA7.05."""
```

File: esmvalcore/cmor/_fixes/cordex/hadrem3_ga7_05.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+from ..fix import Fix
+
+
+class AllVars(Fix):
+    """Fix errors common to all vars."""
+
+    def fix_metadata(self, cubes):
+        """Fix latitude and longitude
```
Comment (valeriupredoi) [inline esmvalcore/cmor/_fixes/cordex/hadrem3_ga7_05.py:9]:
```suggestion
        """Fix latitude and longitude.
```

**Review by valeriupredoi — CHANGES_REQUESTED** [review]
two minor Codacy issues. @jvegasbsc and @zklaus will know about CORDEX, I have very limited knowledge of that dataset unfortunately

**Comment by nhsavage** [comment]
how do we get this PR moving again please? should we do a new PR to check if it will merge to the latest head? What needs to be done to resolve the Codacy Static Code Analysis  problems?

**Comment by zklaus** [comment]
Oh sorry guys, I just started using trello to organize my own ESMValTool work. I did not realize that the "Power-Up" I was using would place a backlink here.

Please disregard

PS: And now I found the switch to avoid that :facepalm:

**Comment by sloosvel** [comment]
This fix got included in #1765

**Pull request closed without merging**